test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/UUIDTest.scala

```diff
@@ -95,21 +95,86 @@ class UUIDTest {
   }
 
   @Test def compareTo(): Unit = {
-    val uuid0101 = new UUID(1L, 1L)
-    val uuid0111 = new UUID(1L, 0x100000001L)
-    val uuid1000 = new UUID(0x100000000L, 0L)
-
-    assertEquals(0, uuid0101.compareTo(uuid0101))
-    assertEquals(0, uuid0111.compareTo(uuid0111))
-    assertEquals(0, uuid1000.compareTo(uuid1000))
-
-    assertTrue(uuid0101.compareTo(uuid0111) < 0)
-    assertTrue(uuid0101.compareTo(uuid1000) < 0)
-    assertTrue(uuid0111.compareTo(uuid1000) < 0)
-
-    assertTrue(uuid0111.compareTo(uuid0101) > 0)
-    assertTrue(uuid1000.compareTo(uuid0101) > 0)
-    assertTrue(uuid1000.compareTo(uuid0111) > 0)
+    /* #4882 `UUID.compareTo()` is known not to match the specification in RFC
+     * 4122. However, the exact algorithm used by the JVM is not publicly
+     * available with a license that we can use. The best we have is the
+     * JavaDoc that says
+     *
+     * > The first of two UUIDs is greater than the second if the most
+     * > significant field in which the UUIDs differ is greater for the first
+     * > UUID.
+     *
+     * We do not know what a "field" is, but we can guess that it is either
```

Comment:
This is a bit misleading IMO. The JavaDoc clearly explains what fields are supposed to be, at least for variant 2:

> The layout of a variant 2 (Leach-Salz) UUID is as follows: The most significant long consists of the following unsigned fields:
> 
>  0xFFFFFFFF00000000 time_low
>  0x00000000FFFF0000 time_mid
>  0x000000000000F000 version
>  0x0000000000000FFF time_hi
>  
> The least significant long consists of the following unsigned fields:
>  0xC000000000000000 variant
>  0x3FFF000000000000 clock_seq
>  0x0000FFFFFFFFFFFF node

Reply:
If those were indeed the fields used in the comparison, the reference comparison would not have worked for a number of the generated pairs. So they must be talking about different fields.

Reply:
I hopefully made it clearer.